apps/webapp/src/fetchers/stream.test.ts

```diff
@@ -0,0 +1,47 @@
+import { beforeEach, describe, expect, test } from 'vitest';
+import { streamToPromise } from './stream';
+
+describe('streamToPromise()', () => {
```

Comment:
See my description inside the `stream.ts` file below — I added these tests due to having to rework the `catch` logic